utils/utils.py

```diff
@@ -1,46 +1,50 @@
-import shutil
```

Comment:
Hi @F-Ali-Khan, thanks for the update. Can I ask why you largely modify this utils while adding the necessary functions for the resolution computation?